internal/profiles/service.go

```diff
@@ -773,22 +772,14 @@ func upsertRuleInstances(
 			return nil, fmt.Errorf("unable to serialize rule params: %w", err)
 		}
 
-		newInstance := db.UpsertRuleInstanceParams{
+		id, err := qtx.UpsertRuleInstance(ctx, db.UpsertRuleInstanceParams{
 			ProfileID:  profileID,
 			RuleTypeID: entityRuleTuple.RuleID,
 			Name:       rule.Name,
 			EntityType: entityType,
-			Def: pqtype.NullRawMessage{
-				RawMessage: def,
-				Valid:      def != nil,
-			},
-			Params: pqtype.NullRawMessage{
-				RawMessage: params,
-				Valid:      params != nil,
-			},
-		}
-
-		id, err := qtx.UpsertRuleInstance(ctx, newInstance)
+			Def:        def,
+			Params:     params,
```

Comment:
I assume that if there's no params defined then json.Marshall returns `{}`? I wasn't sure about that since the previous code had an explicit `params != nil` check. But I guess we'd want to either store `{}` or make sure to on retrieval convert whatever value is stored to `{}` .